llvm-general/src/LLVM/General/Internal/Analysis.hs

```diff
@@ -16,8 +16,9 @@ import LLVM.General.Internal.Coding
 -- | Run basic sanity checks on a 'Module'. Note that the same checks will trigger assertions
 -- within LLVM if LLVM was built with them turned on, before this function can be is called.
 verify :: Module -> ExceptT String IO ()
-verify (Module m) = flip runAnyContT return $ do
+verify m = flip runAnyContT return $ do
   errorPtr <- alloca
-  result <- decodeM =<< (liftIO $ FFI.verifyModule m FFI.verifierFailureActionReturnStatus errorPtr)
+  m' <- readModule m
+  result <- decodeM =<< (liftIO $ FFI.verifyModule m' FFI.verifierFailureActionReturnStatus errorPtr)
```

Comment:
Not strictly necessary, but if you put the prime on the first m (or neither) instead of the second, then you won't need a change and diffs will be smaller.